### ISSUE TITLE
Lane keeping TTS commands

### ISSUE THREAD

**Issue by albertodiazsaez** [osmandapp/OsmAnd#13515]
# 🚀 feature request

### Description

When in a motorway, the TTS only says "Continue for about [distance] on [motorway name]", but the screen shows th# 🚀 feature request

### Description

When in a motorway, the TTS only says "Continue for about [distance] on [motorway name]", but the screen shows the  lanes you have to go to keep on that motorway. The law (here in Spain and I think most of Europe), says that you have to keep going in the right lane, only using the left one for destinations, overtaking, or when the traffic is high, so if the voice does not warn you, its "easy" to take an exit without realizing when the lanes diverge.

I tried in both Spanish and English TTS.

### Describe the solution you'd like

I think the TTS should guide you across lanes, even though you "have to keep going on the same road". In one of my daily commutes, the motorway changes from 3 lanes, to 2 lanes, having the right lane to be an exit, and OSMAnd does not warm me to "Keep on the 2 left lanes to continue on [motorway name]" (The lane issue is not only on motorways, but It's the most problematic I had when navigating with OSMAnd).

The two routing examples I have are this two for the lanes issue:

From: 39.49623 -0.42781 To: 39.49995 -0.39861 (Three lanes, that convert to two, does not warn you).

From: 39.50296 -0.39438 To: 39.49171 -0.43057 (4 lanes, you have to go to the two left ones, does not tell you).

**Comment by vshcherb** [comment]
This issue is related to merge announcement. Similar situation cause you can't stay probably on the same lane we need to warn - unfortunately we don't on which lane now car is standing to handle it nicely though as you mentioned we could assume it sometimes.

**Comment by albertodiazsaez** [comment]
I don't think you need to asume the position of the car, what I would do is to check a change in OSM data for lanes that go through to slight_right/left, and then warn the user to use "two lanes from the left". In this way we avoid warning at every highway exit (because full exit lane should be "slight_right", but warn when the role of the lane changes.

**Comment by chereskata** [comment]
@albertodiazsaez i have also seen, that the line change announcement to take an exit is in fact a turn right command right before the slowdown line diverges from the motorway
<br>
![out2](https://user-images.githubusercontent.com/87526889/188331849-fbca5ecb-c7b2-4901-bdc5-68802f9b107c.jpg) ![out](https://user-images.githubusercontent.com/87526889/188331797-96b55a3e-9aea-42e8-9d7d-fb010b5d79dc.jpg) 

I want to say, that it would be more useful to announce a highway linechange even before the slowdown lane starts. And if the line is tagged incorrectly and only the exit curve is shown, then one could detect, that the slowdown lane has in-fact no touch area with the motorway and calculate an average of like 1km before to give the announce.

**Comment by vshcherb** [comment]
@chereskata please check https://docs.osmand.net/docs/technical/contributions/voice-prompts https://docs.osmand.net/docs/technical/algorithms/voice-prompt-triggering

**Comment by chereskata** [comment]
Thank you, will try it on my next ride. My default speed was 45km/h and i set it to 100km/h now :)
